### ISSUE TITLE
ERROR Could not pass event InventoryClickEvent, InventoryCloseEvent, CraftItemEvent

### ISSUE THREAD

**Issue by radek203** [utarwyn/EnderContainers#33]
```
[23:44:58] [Server thread/ERROR]: Could not pass event InventoryClickEvent to EnderContainers v2.1.0
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1891) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:33) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:10) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13) [server.jar:git-Spigot-dcd1643-e60fc34]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_141]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_141]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:748) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [server.jar:git-Spigot-dcd1643-e60fc34]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_141]
Caused by: java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437) ~[?:1.8.0_141]
	at java.util.HashMap$KeyIterator.next(HashMap.java:1461) ~[?:1.8.0_141]
	at fr.utarwyn.endercontainers.menu.Menus.getMenuFromInventory(Menus.java:91) ~[?:?]
	at fr.utarwyn.endercontainers.menu.Menus.onInventoryClick(Menus.java:58) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor163.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_141]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_141]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	... 15 more
[16:10:53] [Server thread/ERROR]: Could not pass event InventoryCloseEvent to EnderContainers v2.1.0
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.handleInventoryCloseEvent(CraftEventFactory.java:895) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1671) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PacketPlayInCloseWindow.a(PacketPlayInCloseWindow.java:18) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PacketPlayInCloseWindow.a(PacketPlayInCloseWindow.java:1) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13) [server.jar:git-Spigot-dcd1643-e60fc34]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_141]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_141]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:748) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [server.jar:git-Spigot-dcd1643-e60fc34]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_141]
Caused by: java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437) ~[?:1.8.0_141]
	at java.util.HashMap$KeyIterator.next(HashMap.java:1461) ~[?:1.8.0_141]
	at fr.utarwyn.endercontainers.menu.Menus.getMenuFromInventory(Menus.java:91) ~[?:?]
	at fr.utarwyn.endercontainers.menu.Menus.onInventoryClose(Menus.java:77) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor145.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_141]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_141]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	... 16 more
[15:22:39] [Server thread/ERROR]: Could not pass event CraftItemEvent to EnderContainers v2.1.0
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [server.jar:git-Spigot-dcd1643-e60fc34]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1891) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:33) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:10) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13) [server.jar:git-Spigot-dcd1643-e60fc34]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_141]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_141]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:748) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [server.jar:git-Spigot-dcd1643-e60fc34]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [server.jar:git-Spigot-dcd1643-e60fc34]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_141]
Caused by: java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437) ~[?:1.8.0_141]
	at java.util.HashMap$KeyIterator.next(HashMap.java:1461) ~[?:1.8.0_141]
	at fr.utarwyn.endercontainers.menu.Menus.getMenuFromInventory(Menus.java:91) ~[?:?]
	at fr.utarwyn.endercontainers.menu.Menus.onInventoryClick(Menus.java:58) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor163.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_141]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_141]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-dcd1643-e60fc34]
	... 15 more
```
I get such errors ... and sometimes:
```
[13:42:27] [Craft Scheduler Thread - 289/WARN]: [EnderContainers] Plugin EnderContainers v2.1.0 generated an exception while executing task 814745
net.minecraft.server.v1_12_R1.CancelledPacketHandleException: null
```

**Comment by andris155** [comment]
Latest dev. versions fixed this errors, but only compatible 1.14.4.

**Comment by radek203** [comment]
> 
> 
> The latest dev. versions fixed this errors, but only compatible 1.14.4.

where can i download the latest versions?

**Comment by andris155** [comment]
You need compile with apache mave from GitHub source.
